src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -41,18 +41,25 @@ public abstract class AssemblyLoadContext
         [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
         internal static extern void InternalStartProfile(string profile, IntPtr ptrNativeAssemblyLoadContext);
 
-        protected AssemblyLoadContext() : this(false, false)
+        public AssemblyLoadContext() : this(false, false, null)
         {
         }
 
-        protected AssemblyLoadContext(bool isCollectible) : this(false, isCollectible)
+        public AssemblyLoadContext(bool isCollectible) : this(false, isCollectible, null)
         {
         }
 
-        internal AssemblyLoadContext(bool fRepresentsTPALoadContext, bool isCollectible)
+        public AssemblyLoadContext(string name, bool isCollectible = true) : this(false, isCollectible, name)
+        {
+        }
+
+        internal AssemblyLoadContext(bool fRepresentsTPALoadContext, bool isCollectible, string name)
         {
             // Initialize the VM side of AssemblyLoadContext if not already done.
             IsCollectible = isCollectible;
+
+            Name = name;
```

Comment:
TBD should this be a fixed value or an empty string

Reply:
I would definitely leave the variable null.
Currently it's shown as empty string in `ToString`, which is probably OK. I personally would prefer to omit the double quotes in that case completely, but it's just a minor thing.